kratos/utilities/variable_utils.h

```diff
@@ -450,16 +450,42 @@ class KRATOS_API(KRATOS_CORE) VariableUtils
     {
         KRATOS_TRY
 
+        const auto it_cont_begin = rContainer.begin();
+
         #pragma omp parallel for
         for (int k = 0; k< static_cast<int> (rContainer.size()); ++k) {
-            auto it_cont = rContainer.begin() + k;
+            auto it_cont = it_cont_begin + k;
             it_cont->Set(rFlag, rFlagValue);
         }
 
         KRATOS_CATCH("")
 
     }
 
+    /**
+     * @brief Flips a flag over a given container
+     * @param rFlag flag to be set
+     * @param rContainer Reference to the objective container
+     */
+    template< class TContainerType >
+    void ResetFlag(
+        const Flags& rFlag,
+        TContainerType& rContainer
+        )
+    {
+        KRATOS_TRY
+
+        const auto it_cont_begin = rContainer.begin();
+
+        #pragma omp parallel for
```

Comment:
```suggestion
        #pragma omp parallel for firstprivate(it_cont_begin)
```

Reply:
It is not needed when it is const, we already had this discussion in a previous PR